app/src/components/content/ContentTile.spec.ts

```diff
@@ -37,4 +50,30 @@ describe("ContentTile", () => {
 
         expect(wrapper.text()).toContain("Jan 1, 2024");
     });
+
+    it("navigates to the correct route on click", async () => {
+        const routes = [{ path: "/:slug", name: "post", component: SingleContent }];
+
+        const router = createRouter({
+            history: createWebHistory(),
+            routes,
+        });
+
+        const wrapper = mount(ContentTile, {
+            props: {
+                content: mockEnglishContentDto,
+            },
+            global: {
+                plugins: [router],
+            },
+        });
+
+        await wrapper.trigger("click");
+        await router.isReady();
+
+        expect(routePushMock).toHaveBeenCalledWith({
+            name: "post",
+            params: { slug: mockEnglishContentDto.slug },
+        });
+    });
```

Comment:
Nice improvement by testing the actual push from the component. Creating your own router is not needed here (and should indeed almost never be necessary, please think twice if you ever feel like you want to reach for it as it likely means the test is not testing the right thing), so you can remove the following parts:

```suggestion
    it("navigates to the correct route on click", async () => {
        const wrapper = mount(ContentTile, {
            props: {
                content: mockEnglishContentDto,
            },
        });

        await wrapper.trigger("click");

        expect(routePushMock).toHaveBeenCalledWith({
            name: "post",
            params: { slug: mockEnglishContentDto.slug },
        });
    });
```

With these changes the test is still green, as it will now just use the actual route definition of our app, and because we have mocked the `route.push` in the component we can test whether it is called with the right arguments.